cpg_pipes/pipeline/pipeline.py

```diff
@@ -944,7 +916,9 @@ def set_stages(
         # implicit_stages_d: dict[str, Stage] = dict()  # If there are required
         # dependency stages that are not requested explicitly, we are putting them
         # into this dict as `skipped`.
+        depth = 0
         while True:  # Might need several rounds to resolve dependencies recursively.
```

Comment:
Should this include a check for cyclic dependencies?

Reply:
Not here. This code got has grown out of hand a bit and became too complicated. Essentially it's just topological sort here, and I have plans to switch to a more robust algorithm, but that's after I got everything in main.